cmd/kubelet/app/server.go

```diff
@@ -874,6 +874,18 @@ func buildKubeletClientConfig(ctx context.Context, s *options.KubeletServer, nod
 		klog.V(2).Info("Starting client certificate rotation.")
 		clientCertificateManager.Start()
 
+		// we should wait certificate manager to finish client certificate bootstrap, otherwise Kubelet always try to connect
+		// to API Server anonymously before client certificate bootstrap finished.
+		if err := wait.PollImmediateUntil(100*time.Millisecond, func() (done bool, err error) {
```

Comment:
I'm pretty sure blocking here would prevent a self-hosted kubelet (where the kube-apiserver was run as a static pod by the kubelet) from ever initializing. The function explicitly logs "Client rotation is on, will bootstrap in background"

Reply:
see https://github.com/kubernetes/kubernetes/pull/71174/files#diff-e0160f895346669c2e4495acfd858ea7R759